protocol/communities_messenger_test.go

```diff
@@ -3774,3 +3774,167 @@ func (s *MessengerCommunitiesSuite) TestRetrieveBigCommunity() {
+	s.Require().NoError(err)
+	s.Require().Len(requestsToJoin, 1)
+
+	// Make sure the requests are fetched also by community
+	requestsToJoin, err = s.alice.PendingRequestsToJoinForCommunity(community.ID())
+	s.Require().NoError(err)
+	s.Require().Len(requestsToJoin, 1)
+
+	requestToJoin2 := response.RequestsToJoinCommunity[0]
+
+	s.Require().NotNil(requestToJoin2)
+	s.Require().Equal(community.ID(), requestToJoin2.CommunityID)
+	s.Require().NotEmpty(requestToJoin2.ID)
+	s.Require().NotEmpty(requestToJoin2.Clock)
+	s.Require().Equal(requestToJoin2.PublicKey, common.PubkeyToHex(&s.alice.identity.PublicKey))
+	s.Require().Equal(communities.RequestToJoinStatePending, requestToJoin2.State)
+
+	s.Require().Equal(requestToJoin1.ID, requestToJoin2.ID)
+
+	requestToCancel := &requests.CancelRequestToJoinCommunity{ID: requestToJoin1.ID}
+	response, err = s.alice.CancelRequestToJoinCommunity(ctx, requestToCancel)
+	s.Require().NoError(err)
+	s.Require().NotNil(response)
+	s.Require().Len(response.RequestsToJoinCommunity, 1)
+	s.Require().Equal(communities.RequestToJoinStateCanceled, response.RequestsToJoinCommunity[0].State)
+
+	messageState = s.alice.buildMessageState()
+	messageState.CurrentMessageState = &CurrentMessageState{}
+
+	messageState.CurrentMessageState.PublicKey = &s.alice.identity.PublicKey
+
+	statusMessage.TransportLayer.Dst = community.PublicKey()
+
+	requestToJoinCancelProto := &protobuf.CommunityRequestToJoinResponse{
+		CommunityId: community.ID(),
+		Clock:       requestToJoin1.Clock + 1,
+		Accepted:    true,
+	}
+
+	err = s.alice.HandleCommunityRequestToJoinResponse(messageState, requestToJoinCancelProto, &statusMessage)
+	s.Require().NoError(err)
+	aliceJoinedCommunities, err := s.alice.JoinedCommunities()
+	s.Require().NoError(err)
+	// Make sure on Alice side she hasn't joined any communities
+	s.Require().Empty(aliceJoinedCommunities)
+
+	// pull to make sure it has been saved
+	cancelRequestsToJoin, err := s.alice.MyCanceledRequestToJoinForCommunityID(community.ID())
+	s.Require().NoError(err)
+	s.Require().NotNil(cancelRequestsToJoin)
+	s.Require().Equal(cancelRequestsToJoin.State, communities.RequestToJoinStateCanceled)
+
+	s.Require().NoError(err)
+
+	messageState = s.alice.buildMessageState()
+	messageState.CurrentMessageState = &CurrentMessageState{}
+
+	messageState.CurrentMessageState.PublicKey = &s.alice.identity.PublicKey
+
+	statusMessage.TransportLayer.Dst = community.PublicKey()
+
+	requestToJoinResponseProto := &protobuf.CommunityRequestToJoinResponse{
+		Clock:       cancelRequestsToJoin.Clock,
+		CommunityId: community.ID(),
+		Accepted:    true,
+	}
+
+	err = s.alice.HandleCommunityRequestToJoinResponse(messageState, requestToJoinResponseProto, &statusMessage)
+	s.Require().NoError(err)
+	// Make sure alice is NOT a member of the community that she cancelled her request to join to
+	s.Require().False(community.HasMember(s.alice.IdentityPublicKey()))
+	// Make sure there's no PNs for Alice
```

Comment:
The comment should probably be `Make sure there are no AC notifications for Alice`, not `PNs`, which means something else.